workers/generic-worker/d2g_test.go

```diff
@@ -25,18 +28,46 @@ func TestWithValidDockerWorkerPayload(t *testing.T) {
 		Command:    []string{"/bin/bash", "-c", "echo hello world"},
 		Image:      json.RawMessage(imageBytes),
 		MaxRunTime: 30,
+		Artifacts: map[string]dockerworker.Artifact{
+			"testWithoutExpires": {
+				Path: "test/path",
+				Type: "file",
+				// purposely do NOT set Expires
+				// because this is also testing
+				// that the default expires value
+				// is not present in the translated
+				// generic worker task definition
+				// Expires: tcclient.Time{},
+			},
+			"testWithExpires": {
+				Path:    "test/path",
+				Type:    "file",
+				Expires: tcclient.Time(testTime),
+			},
+		},
 	}
 	defaults.SetDefaults(&payload)
 	td := testTask(t)
 
 	switch fmt.Sprintf("%s:%s", engine, runtime.GOOS) {
 	case "multiuser:linux":
 		_ = submitAndAssert(t, td, payload, "completed", "completed")
+		logtext := LogText(t)
+		// tests the default artifact expiry is not present in the
+		// translated task definition
+		if strings.Contains(logtext, "0001-01-01T00:00:00.000Z") {
+			t.Fatal("Was expecting log file to not contain '0001-01-01T00:00:00.000Z'")
+		}
+		// tests the set artifact expiry is present in the
+		// translated task definition
+		if !strings.Contains(logtext, testTime.Format("0001-01-01T00:00:00.000Z")) {
+			t.Fatalf("Was expecting log file to contain '%s'", testTime.Format("0001-01-01T00:00:00.000Z"))
+		}
```

Comment:
I think this needs to be something like

testTime.UTC().Format("2006-01-02T15:04:05.000Z")

Reply:
Alternatively

```go
if text := tcclient.Time(testTime).String(); !strings.Contains(logtext, test) {
  t.Fatalf("Was expecting .... '%s'", text)
}
```